### PR TITLE
Add registry config if not found

### PR DESCRIPTION
**Pull request by bamachrn**

### CODE REVIEW

File: provisions/roles/openshift/tasks/config.yml [hunk 1]
```diff
@@ -1,15 +1,17 @@
 ---
 - name: Enable Docker insecure registry
-  replace: >
-    dest=/etc/sysconfig/docker
+  lineinfile:
+    path=/etc/sysconfig/docker
+    state=present
     regexp="^#*\s*INSECURE_REGISTRY=.*"
```
Comment (dharmit) [inline provisions/roles/openshift/tasks/config.yml]: With the latest version of Docker available on CentOS 7, I saw that this line doesn't exist at all. Not even commented, as was the case earlier. So, I had to leave the `regexp` part altogether. Here's what I did: https://github.com/dharmit/digitalocean-compass-ansible/blob/master/main.yml#L61-L64

Comment (dharmit) [reply provisions/roles/openshift/tasks/config.yml]: So with `state=present`, I think, the `regexp` you've mentioned won't be found and hence the `line` you want to add to the file won't be added.

File: provisions/roles/openshift/tasks/config.yml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-    replace='INSECURE_REGISTRY="--insecure-registry 172.30.0.0/16 --insecure-registry {{ intranet_registry }}"'
+    line='INSECURE_REGISTRY="--insecure-registry 172.30.0.0/16 --insecure-registry {{ intranet_registry }}"'
 
 - name: Enable Docker registry
-  replace: >
-    dest=/etc/sysconfig/docker
+  lineinfile:
+    path=/etc/sysconfig/docker
+    state=present
     regexp="^#*\s*ADD_REGISTRY=.*"
```
Comment (dharmit) [inline provisions/roles/openshift/tasks/config.yml]: Same comment as for `INSECURE_REGISTRY`. There's nothing like `# ADD_REGISTRY` in the `/etc/sysconfig/docker` file.

File: provisions/roles/openshift/tasks/config.yml (hunk 1 above)
Comment (bamachrn) [reply provisions/roles/openshift/tasks/config.yml]: ok but the ansible documentation says it adds if not found. I know it is not there and that is the reason we are hitting the issue

File: provisions/roles/openshift/tasks/config.yml (hunk 2 above)
Comment (bamachrn) [reply provisions/roles/openshift/tasks/config.yml]: sure updating the same

File: provisions/roles/openshift/tasks/config.yml (hunk 1 above)
Comment (dharmit) [reply provisions/roles/openshift/tasks/config.yml]: English isn't my first language and I admit that the documentation is kinda confusing me as well. From the docs:
> For `state=present`, the pattern to replace if found; only the last line found will be replaced. 

Now the `regext` won't be found as required by `the pattern to replace if found`, and as a result, line we're looking to add won't be added. That's how I interpret the statement. You can see in the link I mentioned in first comment that there's no `state` or `regexp` parameter. But, I could be wrong. :smile:

**Comment by bamachrn** [comment]
#dotests

**Comment by bamachrn** [comment]
#dotests-debug

**Pull request merged**